README.md

```diff
@@ -3,161 +3,163 @@
+
+## Best practices
+
+In this section, we provide a few deployment guidelines and then discuss a real-world scenario.
+
+### Deployment guidelines
+
+
+
+*   If there are multiple service operators (a.k.a. [SREs](https://en.wikipedia.org/wiki/Site_reliability_engineering)) deploying different services in a cluster (typically in a medium- or large-size cluster), we recommend creating a separate [namespace](https://kubernetes.io/docs/tasks/administer-cluster/namespaces-walkthrough/) for each SRE team to isolate their access. For example, you could create a "team1-ns" namespace for team1, and "team2-ns" namespace for team2, such that both teams won't be able to access each other's services.
+
+*   If Istio CA is compromised, all its managed keys and certificates in the cluster may be exposed. We *strongly* recommend running Istio CA on a dedicated namespace (for example, istio-ca-ns), which only cluster admins have access to.
 
 ### Example
 
-Let’s consider a 3-tier application with three services: photo-frontend,
-photo-backend, and datastore. Photo-frontend and photo-backend services are
-managed by the photo SRE team while the datastore service is managed by the
-datastore SRE team. Photo-frontend can access photo-backend, and photo-backend
-can access datastore. However, photo-frontend cannot access datastore.
-
-In this scenario, a cluster admin can create 3 namespaces: istio-ca-ns,
-photo-ns, and datastore-ns. Admin has access to all namespaces, and each team
-only has access to its own namespace. The photo SRE team creates 2 service
-accounts to run photo-frontend and photo-backend respectively in namespace
-photo-ns. The datastore SRE team creates 1 service account to run the datastore
-service in namespace datastore-ns. Moreover, we need to enforce the service
-access control in [Istio Mixer](https://github.com/istio/mixer) such that
-photo-frontend cannot access datastore.
-
-In this setup, Istio CA is able to provide key/cert management for all
-namespaces. And we successfully prevent the team from messing up services
-running by other teams.
-
-## Future Work
-
-- Fine-grained authorization and auditing
-- Secure Istio components (Mixer, Istio-Manager, etc.)
-- Inter-cluster service-to-service authentication
-- End-user to service authentication using JWT/OAuth2/OpenID_Connect
-- Support GCP service account and AWS service account
-- Non-http traffic (MySql, Redis, etc.) support
-- Auth info propagation from Envoy to the service
-- Unix domain socket for local communication between service and Envoy
-- Middle proxy support
-- Pluggable key management component
+Let's consider a 3-tier application with three services: photo-frontend, photo-backend, and datastore. Photo-frontend and photo-backend services are managed by the photo SRE team while the datastore service is managed by the datastore SRE team. Photo-frontend can access photo-backend, and photo-backend can access datastore. However, photo-frontend cannot access datastore.
```

Comment:
What is a 3-tier application?

Reply:
? It means we have frontend, backend, datastore.

Reply:
It's not a new change, btw.